sockeye/encoder.py

```diff
@@ -56,11 +56,13 @@ class RecurrentEncoderConfig(Config):
     def __init__(self,
                  rnn_config: rnn.RNNConfig,
                  conv_config: Optional['ConvolutionalEmbeddingConfig'] = None,
-                 reverse_input: bool = False) -> None:
+                 reverse_input: bool = False,
+                 use_fp16: bool = False) -> None:
```

Comment:
I think it would be nicer to rename this to dtype and allow the actualy dtype (imagine we want to train with float64 at some point).

Reply:
+1